Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1

```diff
@@ -285,6 +339,212 @@ function Add-ListOfUsersToDosAdminRole($edwAdminUsers, $connString, $authorizati
     }	
 }
 
+function Add-DosAdminGroup($authUrl, $accessToken, $groupName)
+{
+    try {
+        $group = Add-Group -authUrl $authUrl -name $groupName -source "custom" -accessToken $accessToken
+        return $group
+    }
+    catch {
+        $exception = $_.Exception
+        if ($exception -ne $null -and $exception.Response -ne $null -and $exception.Response.StatusCode.value__ -eq 409) {
+            $group = Get-Group -authorizationServiceUrl $authUrl -name $groupName -accessToken $accessToken
+            Write-Success "$groupName group already exists..."
+            return $group;
+        }
+        else{
+            if ($exception.Response -ne $null) {
+                $error = Get-ErrorFromResponse -response $exception.Response
+                Write-Error "    There was an error adding the Dos Admin group: $error. Halting installation."
+            }
+            throw $exception
+        }
+    }
+}
+
+function Add-DosAdminRoleUsersToDosAdminGroup([GUID]$groupId, $connectionString)
+{
+    $query = "INSERT INTO GroupUsers
+              (CreatedBy, CreatedDateTimeUtc, GroupId, IdentityProvider, SubjectId, IsDeleted)
+              SELECT 'fabric-installer', GETUTCDATE(), @dosAdminGroupId, u.IdentityProvider, ru.subjectid, 0 from RoleUsers ru
+              INNER JOIN Roles r ON r.RoleId = ru.RoleId
+              INNER JOIN Users u ON u.SubjectId = ru.SubjectId
+              INNER JOIN SecurableItems s on r.SecurableItemId = s.SecurableItemId
+              WHERE r.[Name] = 'dosadmin'
+              AND s.[Name] = 'datamarts'
+              AND ru.IsDeleted = 0;"
+    try{
+        Write-Host "Migrating dosadmin role users to Dos Admin group..."
+        Invoke-Sql -connectionString $connectionString -sql $query -parameters @{dosAdminGroupId=$groupId;} | Out-Null
+    }catch{
+        Write-Error $_.Exception
+        throw $_.Exception
+    }
+}
+
+function Remove-UsersFromDosAdminRole($connectionString)
+{
+    $sql = "UPDATE ru 
+            SET ru.IsDeleted = 1,
+                ru.ModifiedBy = 'fabric-installer',
+                ru.ModifiedDateTimeUtc = GETUTCDATE()
+            FROM roleusers ru
+            INNER JOIN roles r ON ru.RoleId = r.RoleId
+            INNER JOIN SecurableItems s on r.SecurableItemId = s.SecurableItemId
+            WHERE r.[Name] = 'dosadmin'
+            AND s.[Name] = 'datamarts';"
```

Comment:
Should we include a `WHERE ru.IsDeleted=0` here so we don't update already deleted records?

Reply:
@bsmithvt - I've updated the query.